types/got/index.d.ts

```diff
@@ -170,6 +172,11 @@ declare namespace got {
      */
     type BeforeRetryHook<Options> = (options: Options, error: GotError, retryCount: number) => any;
 
+    /**
+     * @param error Request or parse error.
+     */
+    type BeforeErrorHook = (error: GotError | ParseError | HTTPError | MaxRedirectsError) => Error | Promise<Error>;
```

Comment:
`GotError`  type is union of other errors: 

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/6e067da3948185d3c01fc53fa33a09af1c98039b/types/got/index.d.ts#L351

Therefore, other error types can be omitted.